app/controllers/api/v1/allocations_controller.rb

```diff
@@ -15,8 +15,14 @@ def index
 
       def create
         allocation = Allocation.new(allocation_attributes)
+        allocation.moves = moves if allocation.valid?
```

Comment:
Could we fully validate and handle the propagated error in the controller with rich validation errors?

Reply:
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api_controller.rb#L17

Reply:
This was initially added to avoid validating on a `move` before actually validating the `allocation` itself. This was happening when we were missing a date, and the `move` validation was taking precedence in the error messages. I have now changed this to only look for the `date` attribute instead of performing the validation twice as that was confusing.